app/services/assignment_service.py

```diff
@@ -249,22 +257,22 @@ def __init__(self, session: Session, instructor: InstructorModel, assignment: As
         self.instructor = instructor
         self.assignment = assignment
 
-    def get_is_available(self) -> bool:
-        if not self.assignment.is_created: return False
+    def _get_is_available(self) -> bool:
+        if not self.assignment.is_published: return False
 
         current_timestamp = self.session.scalar(func.current_timestamp())
         return current_timestamp >= self.assignment.available_date
     
-    def get_is_closed(self) -> bool:
-        if not self.assignment.is_created: return False
+    def _get_is_closed(self) -> bool:
+        if not self.assignment.is_published: return False
 
         current_timestamp = self.session.scalar(func.current_timestamp())
         return current_timestamp > self.assignment.due_date
```

Comment:
here and also in get_is_available I don't think this is the right methodology. Canvas considers an assignment always available to work on if it is both published and is missing either an open or due date. Instead, this will likely just error if the open/due date are unset, since you can't compare a NoneType to an int. The is_created hybrid_property was performing this check for our previously.